### ISSUE TITLE
[V3] Sub-groups and sub-commands aren't inheriting redbot.core.commands subclasses

### ISSUE THREAD

**Issue by Tobotimus** [Cog-Creators/Red-DiscordBot#1784]
In short, the `group()` and `command()` decorators for the `redbot.core.commands.Group` class are not currently implemented, and they need to be in order to use our sub-classed Commands and Groups.

**Comment by palmtree5** [comment]
As an example, take the following eval:

[p]eval
```py
from redbot.core import commands

@commands.command()
async def a(ctx):
    pass

@commands.group()
async def b(ctx):
    pass

@b.command()
async def c(ctx):
    pass

@b.group()
async def d(ctx):
    pass

print(type(a))
print(type(b))
print(type(c))
print(type(d))
```

prints

```py
<class 'redbot.core.commands.commands.Command'>
<class 'redbot.core.commands.commands.Group'>
<class 'discord.ext.commands.core.Command'>
<class 'discord.ext.commands.core.Group'>
```

**Comment by palmtree5** [comment]
Fixed in #1786
